modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -635,7 +635,11 @@ private void validateTransformStatementBody(BLangBlockStmt blockStmt, Set<String
                 }
 
                 // a variable defined in transform scope is a inner variable
-                if (variable.expr.getKind() != NodeKind.LITERAL) {
+                if ((variable.expr.getKind() != NodeKind.LITERAL)
```

Comment:
Can we please factor this out and create another private method to do this. Also taking it to the next step, we need a general util method somewhere to list out all the "literal" data types, so there would be a central place to get that information, rather than duplicating this logic everywhere when someone needs similar functionality.

Reply:
Fixed with https://github.com/ballerinalang/ballerina/pull/3672/commits/db487e6f549f811b3cbbf2f5bd9491273e5abb56